src/main/java/com/shibumi/MaximumPathSum.java

```diff
@@ -29,8 +34,30 @@ public int calculateMaxSum() {
                 memo[i][j] = value;
             }
         }
-        //print(memo);
-        return memo[matrix.length-1][matrix[0].length-1];
+        return memo;
+    }
+
+    public String findMaximumPath() {
+        int[][] maxValuesMatrix = calculateValuesMatrix();
+        return findPath(maxValuesMatrix);
+    }
+
+    private String findPath(int[][] memo) {
+        String path="";
+        int row=memo.length-1, col=memo[0].length-1;
```

Comment:
I love that you are using meaningful iterator names rather than just "i" or "j" here

Reply:
Thanks, it really helped make things more clear, I'm curious what your solution would've been? does it include optimizing space?